l10n_es_aeat_mod347/models/mod347.py

```diff
@@ -0,0 +1,852 @@
+        for item in self:
+            # Browse partner record lines to check if all are correct (all
+            # fields filled)
+            partner_errors = []
+            for partner_record in item.partner_record_ids:
+                if not partner_record.check_ok:
+                    partner_errors.append(
+                        _("- %s (%s)") %
+                        (partner_record.partner_id.name,
+                         partner_record.partner_id.id))
+            real_state_errors = []
+            for real_estate_record in item.real_estate_record_ids:
+                if not real_estate_record.check_ok:
+                    real_state_errors.append(
+                        _("- %s (%s)") %
+                        (real_estate_record.partner_id.name,
+                         real_estate_record.partner_id.id))
+            error = _("Please review partner and real estate records, "
+                      "some of them are in red color:\n\n")
+            if partner_errors:
+                error += _("Partner record errors:\n")
+                error += '\n'.join(partner_errors)
+                error += '\n\n'
+            if real_state_errors:
+                error += _("Real estate record errors:\n")
+                error += '\n'.join(real_state_errors)
+            if partner_errors or real_state_errors:
+                raise exceptions.ValidationError(error)
+        return super(L10nEsAeatMod347Report, self).button_confirm()
+
+    @api.multi
+    def btn_list_records(self):
+        return {
+            'domain': "[('report_id','in'," + str(self.ids) + ")]",
+            'name': _("Partner records"),
+            'view_mode': 'tree,form',
+            'view_type': 'form',
+            'res_model': 'l10n.es.aeat.mod347.partner_record',
+            'type': 'ir.actions.act_window',
+        }
+
+    def _account_move_line_domain(self, taxes):
+        """Return domain for searching move lines.
+
+        :param: taxes: Taxes to look for in move lines.
+        """
+        return [('partner_id.not_in_mod347', '=', False),
+                ('date', '>=', self.date_start),
+                ('date', '<=', self.date_end),
+                '|',
+                ('tax_ids', 'in', taxes.ids),
+                ('tax_line_id', 'in', taxes.ids)]
+
+    @api.model
+    def _get_taxes(self, map):
+        tax_obj = self.env['account.tax']
+        # Obtain all the taxes to be considered
+        tax_templates = map.mapped('tax_ids').mapped('description')
+        if not tax_templates:
+            raise exceptions.Warning(_('No Tax Mapping was found'))
+        # search the account.tax referred to by the template
+        taxes = tax_obj.search(
+            [('description', 'in', tax_templates),
+             ('company_id', 'child_of', self.company_id.id)])
+        return taxes
+
+    def _create_partner_records(self, key, map):
+        map_a = self.env.ref(map)
+        taxes_a = self._get_taxes(map_a)
```

Comment:
¿Por que usáis los IVAs? De hecho, no es relevante que IVA se esten usando. Nosatros no tenemos IVA (empresa de servicios médicos) y debemos presentar igual el 347 con los clientes que superen dicho importe. En mi opinión debería incluirse todos los receivables.

Reply:
Se usa los impuestos porque el importe de los 3000 es sin tener en cuenta el IRPF. Entonces todo eso se necesita saber. En tu caso, si te has creado nuevos impuestos, tendrás que mapearlos como válidos para el 347.